src/cs/Ssh.Tcp/Services/PortForwardingService.cs

```diff
@@ -893,26 +930,26 @@ await remoteConnector.OnChannelOpeningAsync(request, cancellation)
 		return channel;
 	}
 
-	internal void AddChannelForwarder(ChannelForwarder channelForwarder)
+	internal void AddStreamForwarder(StreamForwarder streamForwarder)
 	{
 		if (this.disposed)
 		{
-			channelForwarder.Dispose();
+			streamForwarder.Dispose();
 		}
 		else
 		{
-			lock (this.channelForwarders)
+			lock (this.streamForwarders)
```

Comment:
The seems to a potential for a race condition here: if `this.disposed` is false when we check for that and then `Dispose()` happens after the `if (this.disposed)` and before `lock (this.streamForwarders)`, the stream forwarder will be added to `this.streamForwarders` and never disposed.

Reply:
Fixed, for all the disposable collections in this class.